### ISSUE TITLE
ParentNodeExtensions.PrependNodes throws IndexOutOfRangeException if it gets an empty array

### ISSUE THREAD

**Issue by thorn0** [AngleSharp/AngleSharp#59]
``` c#
var document = DocumentBuilder.Html("");
var nodeArray = DocumentBuilder.HtmlFragment("<br>").ToArray();
document.Body.Prepend(nodeArray);
Console.WriteLine(document.ToHtml());
```

prints

``` html
<html><head></head><body><html><head></head><body><br></body></html></body></html>
```

I expected to get:

``` html
<html><head></head><body><br></body></html>
```

**Comment by thorn0** [comment]
Ah, sorry, my bad. I just didn't understand what `DocumentBuilder.HtmlFragment` does.

**Comment by FlorianRappl** [comment]
No problemo - the fragment is mainly used internally, e.g., when `InnerHtml` or `OuterHtml` properties are being set. The context (not given in the example above) determines what mode to enter.

But I see what you want to achieve. I'll provide some handy helper methods for scenarios like that.

**Comment by thorn0** [comment]
Yep, I need to convert some markup to a node. It might be a tag or a comment, so straight-forward code like `DocumentBuilder.Html(markup).Body.FirstChild` doesn't work as the comment doesn't go to the body. Would be great to have jQuery-like methods that accept strings like `node.prepend(stringWithMarkup)`.

**Comment by thorn0** [comment]
BTW, when I pass an empty array to `ParentNodeExtensions.PrependNodes`, it throws `IndexOutOfRangeException`.

**Comment by FlorianRappl** [comment]
Yep, should and will be fixed.

**Comment by FlorianRappl** [comment]
I've added some methods that should be helpful for scenarios as the original one.

Basically these allow you to write code such as:

```
using AngleSharp.Linq; // extensions ...
// ...
var document = DocumentBuilder.Html("");
document.QuerySelectorAll("body").Prepend("<br>");
Console.WriteLine(document.ToHtml());
```

Hope that helps!

**Comment by thorn0** [comment]
Oh, thanks, looks great. But your code example doesn't compile as these extension methods are for `IEnumerable<IElement>`, not `IElement`.

**Comment by FlorianRappl** [comment]
Oops! You are right. My mind compiler is fucked up. Well, something is needed (but not provided by AngleSharp, at least publicly) to convert a `T` to an `IEnumerable<T>`. I adjusted the example above.

Most of the extensions in AngleSharp only work with an enumerable. I tried to make this close to jQuery, but I only add methods that are not provided by LINQ directly (hence methods like Skip, Select, Count, etc. are not duplicated with other names).
